app/services/usps_exporter.rb

```diff
@@ -0,0 +1,50 @@
+class UspsExporter
+  OTP_MAX_VALID_DAYS = 30
+
+  def initialize(csv_file_path)
+    @csv_file_path = csv_file_path
+  end
+
+  def run
+    CSV.open(csv_file_path, 'wb', col_sep: '|') do |csv|
+      make_csv(csv)
+    end
+    clear_entries
+  end
+
+  private
+
+  attr_reader :csv_file_path
+
+  def make_csv(csv)
+    entries.map(&:decrypted_entry).each do |entry|
+      csv << make_entry_row(entry)
+    end
+  end
+
+  def entries
+    @entries ||= UspsConfirmation.all
+  end
+
+  def clear_entries
+    UspsConfirmation.where(id: entries.map(&:id)).destroy_all
+  end
+
+  # rubocop:disable MethodLength, AbcSize
+  def make_entry_row(entry)
+    now = Time.zone.now
+    due = now + OTP_MAX_VALID_DAYS.days
```

Comment:
putting `now` and `due` into their own methods won't eliminate the rubocode errors.
```
Offenses:

app/services/usps_exporter.rb:33:3: C: Assignment Branch Condition size for make_entry_row is too high. [16/15]
  def make_entry_row(entry)
  ^^^
app/services/usps_exporter.rb:33:3: C: Method has too many lines. [11/10]
  def make_entry_row(entry) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^

519 files inspected, 2 offenses detected
make: *** [lint] Error 1
[peterkarman@XCOH2J-3BBFVH8:~/src/identity-idp]$ git diff
diff --git a/app/services/usps_exporter.rb b/app/services/usps_exporter.rb
index 87eff908..835571b4 100644
--- a/app/services/usps_exporter.rb
+++ b/app/services/usps_exporter.rb
@@ -30,10 +30,7 @@ class UspsExporter
     UspsConfirmation.where(id: entries.map(&:id)).destroy_all
   end
 
-  # rubocop:disable MethodLength, AbcSize
   def make_entry_row(entry)
-    now = Time.zone.now
-    due = now + OTP_MAX_VALID_DAYS.days
     [
       "#{entry.first_name} #{entry.last_name}",
       entry.address1,
@@ -46,5 +43,12 @@ class UspsExporter
       "#{due.strftime('%B %e')},#{due.year}",
     ]
   end
-  # rubocop:enable MethodLength, AbcSize
+
+  def now
+    @_now ||= Time.zone.now
+  end
+
+  def due
+    @_due ||= now + OTP_MAX_VALID_DAYS.days
+  end
 end
```